android/titanium/src/java/org/appcelerator/titanium/proxy/ActivityProxy.java

```diff
@@ -82,6 +82,14 @@ protected Activity getWrappedActivity()
 		return TiApplication.getInstance().getRootActivity();
 	}
 
+	@Kroll.method
+	public void recreate()
+	{
+		if (activity.get() != null) {
```

Comment:
there *could* be a timing issue here since you're not assigning the result to a temp variable. It's extremely unlikely, but the reference could become null after this check but before you call it again to call `recreate()` on it.

Reply:
Chris is right. There is a race condition here. You should do it like this...
```java
Activity activity = this.activity.get();
if (activity != null) {
	activity.recreate();
}
```

Assigning it to a local hard reference like the above keeps the object alive and makes it safe to use after the null check.

Reply:
Also, are you sure calling the `recreate()` method is a good idea? My understanding is that it will destroy the activity and then create a new instance. You would typically only do this if the activity loads its UI from XML. But since we create the UI dynamically via code, all of the displayed views will be lost and the new activity will be blank. Or is our "AndroidManifest.xml" `configChanges` attribute overriding this behavior and it's calling the `Activity.onConigurationChanged()` method instead?

And requiring the app developer to call `recreate()` for every existing activity window would be rather cumbersome too. I'd rather not expose this method in JavaScript and we invoke this method for them automatically for every activity (assuming we need this method at all). This means we create our own event listener handling for this or see if we can leverage the activity `onConfigurationChange()` method somehow.

What about the `ActionBar` title text? By default, the text comes from "strings.xml" and uses the same title resource used to display the app name under the app icon. Things like this tend to only be re-fetched when inflated from XML on Google's end, and that may not happen here. That's partially why I had to create that Java `TiActionBarHandler` class and do it myself.